app/models/spree_klaviyo/newsletter_subscriber_decorator.rb

```diff
@@ -0,0 +1,47 @@
+module SpreeKlaviyo
+  module NewsletterSubscriberDecorator
+    def self.prepended(base)
+      base.store_accessor :private_metadata, :klaviyo_subscribed
+
+      base.after_commit :subscribe_to_klaviyo, if: :verified_at_previously_changed?
+
+      def base.needs_verification?
+        false
+      end
+
+      def base.subscribe(email:, user: nil)
+        Spree::Newsletter::Subscribe.new(email: email, current_user: user).call.tap do |subscriber|
```

Comment:
Can we switch from using `tap` to just a regular flow? It's confusing with the `if` statements inside and calling other services. Same with `Subscribe` and `Unsubscribe` services.